src/router/socketRouter.ts

```diff
@@ -0,0 +1,29 @@
+import {Socket} from 'socket.io';
+import {handleConnect, handleDisconnect, handleError} from '../controller/connectionController.js';
+import { Game, User } from '../entity/index.js';
+
+export type NopeSocket = Socket<ClientToServerEvents, ServerToClientEvents, InterServerEvents, SocketData>;
+
+export interface ServerToClientEvents {
+    error: (message:string) => void;
+    gameInvite :() => void;
+  }
+
+export interface ClientToServerEvents {
+    joinGame: (gameId: string) => void;
+}
+
+/* eslint-disable @typescript-eslint/no-empty-interface*/ // no interserverevents in this project
+export interface InterServerEvents {
+}

```

Comment:
Wenn wir die nicht haben, sollte dann solch ein Interface definiert sein?
Nutzen wir das noch?

Reply:
Interface ist doch nötig für die eigene Typdefinition (siehe [Socket.io Doku](https://socket.io/docs/v4/typescript/#custom-types-for-each-namespace)).